src/audio/google/google_rtc_audio_processing.c

```diff
@@ -83,30 +110,131 @@ void GoogleRtcFree(void *ptr)
 	return rfree(ptr);
 }
 
-#if CONFIG_IPC_MAJOR_4
-static void google_rtc_audio_processing_params(struct processing_module *mod)
+static ALWAYS_INLINE float s16_to_float(const char *ptr)
 {
-	struct google_rtc_audio_processing_comp_data *cd = module_get_private_data(mod);
-	struct sof_ipc_stream_params *params = mod->stream_params;
-	struct comp_buffer *sinkb, *sourceb;
-	struct list_item *source_list;
-	struct comp_dev *dev = mod->dev;
+	float scale = -(float)SHRT_MIN;
+	float x = *(int16_t *)ptr;
 
-	ipc4_base_module_cfg_to_stream_params(&mod->priv.cfg.base_cfg, params);
-	component_set_nearest_period_frames(dev, params->rate);
+	return (1.0f / scale) * x;
+}
+
+static ALWAYS_INLINE void float_to_s16(float x, char *dst)
+{
+	float scale = -(float)SHRT_MIN;
+	float min = -1.0f;
+	float max = 1.0f - 1.0f / scale;
```

Comment:
That's a build-time constant.  There's no division emitted by the compiler, the loop is actually very tight; some earlier disassembly was posted above, it's gotten better since.